adapter/internal/api/apis_process.go

```diff
@@ -0,0 +1,212 @@
+		loggers.LoggerAPI.Debugf("openAPI file : %v", fileName)
+		swaggerJsn, conversionErr := utills.ToJSON(fileContent)
+		if conversionErr != nil {
+			loggers.LoggerAPI.Errorf("Error converting api file to json: %v", conversionErr.Error())
+			return conversionErr
+		}
+		apiProject.OpenAPIJsn = swaggerJsn
+
+		// Interceptor certs
+	} else if strings.Contains(fileName, interceptorCertDir+string(os.PathSeparator)) &&
+		(strings.HasSuffix(fileName, crtExtension) || strings.HasSuffix(fileName, pemExtension)) {
+		if !tlsutils.IsPublicCertificate(fileContent) {
+			loggers.LoggerAPI.Errorf("Provided interceptor certificate: %v is not in the PEM file format. ", fileName)
+			return errors.New("interceptor certificate Validation Error")
+		}
+		apiProject.InterceptorCerts = append(apiProject.InterceptorCerts, fileContent...)
+		apiProject.InterceptorCerts = append(apiProject.InterceptorCerts, newLineByteArray...)
+
+		// Endpoint certs
+	} else if strings.Contains(fileName, endpointCertDir+string(os.PathSeparator)) {
+		if strings.Contains(fileName, endpointCertFile) {
+			epCertJSON, conversionErr := utills.ToJSON(fileContent)
+			if conversionErr != nil {
+				loggers.LoggerAPI.Errorf("Error converting %v file to json: %v", fileName, conversionErr.Error())
+				return conversionErr
+			}
+			endpointCertificates := &model.EndpointCertificatesDetails{}
+			err := json.Unmarshal(epCertJSON, endpointCertificates)
+			if err != nil {
+				loggers.LoggerAPI.Error("Error parsing content of endpoint certificates: ", err)
+			} else if endpointCertificates != nil && len(endpointCertificates.Data) > 0 {
+				for _, val := range endpointCertificates.Data {
+					apiProject.EndpointCerts[val.Endpoint] = val.Certificate
+				}
+			}
+		} else if strings.HasSuffix(fileName, crtExtension) || strings.HasSuffix(fileName, pemExtension) {
+			if !tlsutils.IsPublicCertificate(fileContent) {
+				loggers.LoggerAPI.Errorf("Provided certificate: %v is not in the PEM file format. ", fileName)
+				// TODO: (VirajSalaka) Create standard error handling mechanism
+				return errors.New("certificate Validation Error")
+			}
+
+			if fileNameArray := strings.Split(fileName, string(os.PathSeparator)); len(fileNameArray) > 0 {
+				certFileName := fileNameArray[len(fileNameArray)-1]
+				apiProject.UpstreamCerts[certFileName] = fileContent
+			}
+		}
+
+		// api.yaml or api.json
+	} else if (strings.Contains(fileName, apiYAMLFile) || strings.Contains(fileName, apiJSONFile)) &&
+		!strings.Contains(fileName, openAPIDir) {
+		apiYaml, err := ExtractAPIYaml(fileContent)
+		if err != nil {
+			loggers.LoggerAPI.Errorf("Error while reading %v. %v", fileName, err)
+			return errors.New("Error while reading api.yaml or api.json")
+		}
+		apiProject.APIYaml = apiYaml
+	}
+	return nil
+}
+
+// ExtractAPIYaml returns an APIYaml struct after reading and validating api.yaml or api.json
+func ExtractAPIYaml(fileContent []byte) (apiYaml model.APIYaml, err error) {
```

Comment:
this can be moved to `api_yaml.go`. probably as a constructor `NewAPIYaml(fileContent []byte)`

Reply:
Makes sense. Moving the method to `api_yaml.go` and renaming the method to `UnmarshalAndValidateAPIYaml`